fedimint-core/src/task.rs

```diff
@@ -161,6 +161,45 @@ impl TaskGroup {
         rx
     }
 
+    /// This is a version of [`Self::spawn`] that uses less noisy logging level
+    ///
+    /// Meant for tasks that are spawned often enough to not be as interesting.
+    pub fn spawn_silent<Fut, R>(
+        &self,
+        name: impl Into<String>,
+        f: impl FnOnce(TaskHandle) -> Fut + MaybeSend + 'static,
+    ) -> oneshot::Receiver<R>
+    where
+        Fut: Future<Output = R> + MaybeSend + 'static,
+        R: MaybeSend + 'static,
+    {
+        // Unfortunately this is exact copy of [`Self::spawn`] because logging levels
+        // are static, and can't be parametrized at runtime, AFAIA. --dpc
+        let name = name.into();
+        let mut guard = TaskPanicGuard {
+            name: name.clone(),
+            inner: self.inner.clone(),
+            completed: false,
+        };
+        let handle = self.make_handle();
+
+        let (tx, rx) = oneshot::channel();
+        let handle = crate::runtime::spawn(&name, {
+            let name = name.clone();
+            async move {
+                // if receiver is not interested, just drop the message
+                trace!(target: LOG_TASK, "Starting task {name}");
+                let r = f(handle).await;
+                trace!(target: LOG_TASK, "Finished task {name}");
+                let _ = tx.send(r);
```

Comment:
Could the two functions be unified by using the `event!` macro here (takes log level as argument)? Looks like quite some copy-pasta. Instead we could have a `spawn_with_log_level` fn that is then used by both `spawn` and `spawn_silent`.

Reply:
> takes log level as argument

I tried and logging level needs to be a static/constant, due to the way logging is implemented for performance (basically a single static record in a bin's static section). I know because I implemented it the same way in `slog`, and that's the same way it was implemented in `log`. :D

https://github.com/tokio-rs/tracing/blob/b4868674ba73f3963b125ec38b57efaadc714d90/tracing/src/macros.rs#L590

Reply:
Well... I guess I could just copy&paste the logging in an `if` of some kind instead of the whole function...

OK, I'll do that.

Reply:
Could it be a const generic argument?